internal/controllers/vgmanager/lvmd/lvmd.go

```diff
@@ -16,57 +25,90 @@ type ThinPoolConfig = lvmd.ThinPoolConfig
-
-func DefaultConfigurator() FileConfig {
-	return NewFileConfigurator(DefaultFileConfigPath)
-}
-
-func NewFileConfigurator(path string) FileConfig {
-	return FileConfig{path: path}
+func NewFileConfigurator(client client.Client, namespace string) FileConfig {
+	return FileConfig{Client: client, Namespace: namespace}
 }
 
 type Configurator interface {
-	Load() (*Config, error)
-	Save(config *Config) error
-	Delete() error
+	Load(ctx context.Context) (*Config, error)
+	Save(ctx context.Context, config *Config) error
+	Delete(ctx context.Context) error
 }
 
 type FileConfig struct {
-	path string
+	client.Client
+	Namespace string
 }
 
-func (c FileConfig) Load() (*Config, error) {
-	cfgBytes, err := os.ReadFile(c.path)
-	if os.IsNotExist(err) {
+func (c FileConfig) Load(ctx context.Context) (*Config, error) {
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.LVMDConfigMapName,
+			Namespace: c.Namespace,
+		},
+	}
+	err := c.Client.Get(ctx, client.ObjectKeyFromObject(cm), cm)
+	if k8serrors.IsNotFound(err) {
 		// If the file does not exist, return nil for both
 		return nil, nil
 	} else if err != nil {
-		return nil, fmt.Errorf("failed to load config file %s: %w", c.path, err)
-	} else {
-		config := &Config{}
-		if err = yaml.Unmarshal(cfgBytes, config); err != nil {
-			return nil, fmt.Errorf("failed to unmarshal config file %s: %w", c.path, err)
-		}
-		return config, nil
+		return nil, fmt.Errorf("failed to get ConfigMap %s: %w", cm.Name, err)
 	}
+
+	config := &Config{}
+	if err = yaml.Unmarshal([]byte(cm.Data["lvmd.yaml"]), config); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal config file: %w", err)
+	}
+	return config, nil
 }
 
-func (c FileConfig) Save(config *Config) error {
-	out, err := yaml.Marshal(config)
+func (c FileConfig) Save(ctx context.Context, config *Config) error {
+	logger := log.FromContext(ctx)
+	// remove the old config file if it still exists
+	_, err := os.ReadFile(constants.LVMDDefaultFileConfigPath)
 	if err == nil {
-		err = os.WriteFile(c.path, out, 0600)
+		if err = os.Remove(constants.LVMDDefaultFileConfigPath); err != nil {
+			logger.Info("failed to remove the old lvmd config file", "filePath", constants.LVMDDefaultFileConfigPath, "error", err)
+		}
+	}
+	out, err := yaml.Marshal(config)
+	if err != nil {
+		return fmt.Errorf("failed to marshal config file: %w", err)
+	}
+
+	cm := &corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      constants.LVMDConfigMapName,
+			Namespace: c.Namespace,
+		},
 	}
+	result, err := ctrl.CreateOrUpdate(ctx, c.Client, cm, func() error {
+		cm.Data = map[string]string{"lvmd.yaml": string(out)}
+		return nil
+	})
 	if err != nil {
-		return fmt.Errorf("failed to save config file %s: %w", c.path, err)
+		return fmt.Errorf("failed to apply ConfigMap %s: %w", cm.GetName(), err)
 	}
+
+	logger.Info("ConfigMap applied to cluster", "operation", result, "name", cm.GetName())
```

Comment:
this will log on every reconcile, is that too much for us?

Reply:
It will actually be logged when there is a change in the config. But, still, I removed it. Logging the failure should suffice here.